pypesto/objective.py

```diff
@@ -81,6 +80,18 @@ def __init__(self, fun,
         self.res = res
         self.sres = sres
 
+        self.dim = dim
```

Comment:
we had discussed with Jan to have dim only in the problem description, because sometimes an objective can be called with different argument lengths, and because we have this whole fixed parameters stuff. i am currently also working on sth related, and introduced in Problem a full_dim and a dim field which can be filled in diverse ways.
would there be a reason to have dim in objective with amici?